save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProject.kt

```diff
@@ -0,0 +1,60 @@
+package org.cqfn.save.preprocessor.controllers
+
+import org.cqfn.save.preprocessor.Response
+import org.cqfn.save.repository.GitRepository
+
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.transport.CredentialsProvider
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+import reactor.core.publisher.Mono
+import java.io.File
+
+/**
+ * A Spring controller for git project downloading
+ */
+@RestController
+class DownloadProject {
+    private val log = LoggerFactory.getLogger(DownloadProject::class.java)
+
+    @Value("\${save.repository}")
+    private lateinit var volumes: String
+
+    /**
+     * @param gitRepository - Dto of repo information to clone
+     * @return response entity with text
+     */
+    @Suppress("TooGenericExceptionCaught")
+    @PostMapping(value = ["/upload"])
+    fun upload(@RequestBody gitRepository: GitRepository): Response {
+        val urlHash = gitRepository.url.hashCode()
+        val tmpDir = File("$volumes/$urlHash")
+        tmpDir.deleteRecursively()
+        log.info("For ${gitRepository.url} repository: dir $urlHash was created")
```

Comment:
Why do you say that it has been created when you have just deleted it? You should call `mkdirs` for it to be created